### DISCUSSION TITLE
Support AAD as the authentication feature for AKS

### DISCUSSION THREAD

**Discussion by haminlee** [eclipse-theia/theia-cloud#159] [Ideas]
Hello team,

The current theia-cloud on AKS supports Keycloak as the authentication feature. Is there any optional way to use AAD or other authentication server instead of Keycloak?

It seems the keycloak is the only option we can take from the log below.

20:34:20.126 [main] INFO  org.eclipse.theia.cloud.operator.Main - [init] Parsing args: keycloak false
20:34:20.129 [main] INFO  org.eclipse.theia.cloud.operator.Main - [init] Parsing args: eagerStart false
20:34:20.131 [main] INFO  org.eclipse.theia.cloud.operator.Main - [init] Parsing args: cloudProvider GKE
20:34:20.132 [main] INFO  org.eclipse.theia.cloud.operator.Main - [init] Parsing args: bandwidthLimiter K8SANNOTATION
20:34:20.133 [main] INFO  org.eclipse.theia.cloud.operator.Main - [init] Parsing args: appId asdfghjkl
20:34:20.133 [main] INFO  org.eclipse.theia.cloud.operator.Main - [init] Parsing args: serviceUrl https://service.20.119.119.99.nip.io
20:34:20.134 [main] INFO  org.eclipse.theia.cloud.operator.Main - [init] Parsing args: sessionsPerUser 1
20:34:20.135 [main] INFO  org.eclipse.theia.cloud.operator.Main - [init] Using org.eclipse.theia.cloud.operator.di.DefaultTheiaCloudOperatorModule as DI module
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".

**Comment by jfaltermeier** [comment]
Hi, 

from a technical point of view our Rest-API is built with Quarkus which supports the Open ID Connect protocol. 
The reverse proxy for the running Theia Applications is OAuth2-Proxy, which also supports a wide range of auth providers (https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider). 
So Theia Cloud is not locked into Keycloak. 

In practice we only support Keycloak at the moment, mainly because there are higher prioritised tickets. Keycloak can be installed in any cluster and may be configured to delegate to different auth providers as well. 
So you could check how to use AAD as a Keycloak identity provider. 

There should be no technical blocker to integrate different auth provider directly in the future however and contributions are welcome.

**Reply by haminlee** [reply]
Thanks so much for confirming the current feature!

Yes, I can use AAD through keycloak identity provider instead.
